borg/_hashindex.c

```diff
@@ -400,6 +435,15 @@ hashindex_set(HashIndex *index, const void *key, const void *value)
         while(!BUCKET_IS_EMPTY(index, idx) && !BUCKET_IS_DELETED(index, idx)) {
             idx = (idx + 1) % index->num_buckets;
         }
+        if(BUCKET_IS_EMPTY(index, idx)){
+            index->num_empty--;
```

Comment:
empty = never used. deleted = tombstone.

i am only counting empty buckets in num_empty, why should i increase it when inserting on a tombstone? tombstones/deleted get reduced by 1 due to that (but i do not count tombstones), but empty do not change.